tests/scripts/test_new_matrix_user.py

```diff
@@ -52,16 +52,16 @@ def post(url, json=None, verify=None):
         out = []
         err_code = []
 
-        request_registration(
-            "user",
-            "pass",
-            "matrix.org",
-            "shared",
-            admin=False,
-            requests=requests,
-            _print=out.append,
-            exit=err_code.append,
-        )
+        with patch("synapse._scripts.register_new_matrix_user.requests", requests):
```

Comment:
If you'd like you can patch the entire class and a mock will get passed into each test method IIRC.

Reply:
Good to know! I think I default to applying patches to as small a scope as possible, but I can see that it'd save a lot of boilerplate.